compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -41,6 +41,25 @@ object Parsers {
     def nonePositive: Boolean = parCounts forall (_ <= 0)
   }
 
+  /** Checks whether `t` is a wildcard type.
+   *  If it is, returns the [[Position]] where the wildcard occurs.
+   */
+  @tailrec
+  final def findWildcardType(t: Tree): Option[Position] = t match {
```

Comment:
You could avoid using an `Option` here by returning `NoPosition`. You can then call `exists` on `Position`.

Reply:
Makes sense, but is that a good idea even if the existing code used `Option[Position]`?

Reply:
The existing code is not always right :)

Reply:
I guess it was a defensive move to accommodate the fact that the returned position could be NoPosition. So, yes, I'd leave it as is.

Reply:
Actually, self-answer: probably yes, that's the point of `NoPosition`, plus it avoids allocations.